opta/core/cloud_provider.py

```diff
@@ -0,0 +1,52 @@
+import boto3
+from botocore.config import Config
+from botocore.exceptions import ClientError
+
+from opta.core.terraform2 import TerraformFile
+from opta.layer2 import Layer
+
+
+class CloudProvider:
+    def state_storage(self, layer: Layer) -> str:
+        return f"opta-tf-state-{layer.org_name}-{layer.name}"
+
+    def verify_storage(self, layer: Layer) -> bool:
+        raise NotImplementedError()
+
+    def add_provider_config(self, file: TerraformFile, layer: Layer) -> None:
+        raise NotImplementedError()
+
+
+class AWSProvider(CloudProvider):
+    def verify_storage(self, layer: Layer) -> bool:
```

Comment:
I could not find where was this function used

Reply:
It's currently unused, but will be used shortly once remote state is implemented for module API.